src/django_pg_migration_tools/operations.py

```diff
@@ -1400,6 +1442,69 @@ def describe(self) -> str:
         )
 
 
+class SaferAddFieldOneToOne(operation_fields.AddField):
+    """
+    Django's OneToOneField behaves the same way as a ForeignKey. Except that
+    as an addition, the FK field also contains a unique constraint.
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        name: str,
+        field: models.OneToOneField[models.Model],
+        preserve_default: bool = True,
+    ) -> None:
+        if field.primary_key:
+            raise ValueError("SaferAddFieldOneToOne does not support primary_key=True.")
```

Comment:
👍  This is sensible. The only time `primary_key` would be `True` is if the field is being added when the model is added as part of the implementation of multi-table inheritance. And since the DB table doesn't exist yet at that point, there's no need for a "safer" operation.